src/System.Private.ServiceModel/tests/Common/src/TestTypes.cs

```diff
@@ -702,10 +709,16 @@ public Task<Guid> ServicePingCallback(Guid guid)
 
     public Task<Guid> ServicePingFaultCallback(Guid guid)
     {
-        throw new FaultException<FaultDetail>(
-          new FaultDetail("Throwing a Fault Exception from the Callback method."), 
-          new FaultReason("Reason: Testing FaultException returned from Duplex Callback"), 
-          new FaultCode("ServicePingFaultCallback"),
-          "http://tempuri.org/IWcfDuplexTaskReturnCallback/ServicePingFaultCallback");
+        var fault = new FaultException<FaultDetail>(
+            new FaultDetail("Throwing a Fault Exception from the Callback method."),
+            new FaultReason("Reason: Testing FaultException returned from Duplex Callback"),
+            new FaultCode("ServicePingFaultCallback"),
+            "http://tempuri.org/IWcfDuplexTaskReturnCallback/ServicePingFaultCallbackFaultDetailFault");
+        if (_wrapExceptionInTask)
+        {
+            return Task.FromException<Guid>(fault);
```

Comment:
I'll change this to use a TaskCompletionSource instead. Task.FromException only appeared (externally anyway) in 4.6 on Desktop so is new.